sympy/functions/elementary/tests/test_exponential.py

```diff
@@ -484,6 +484,25 @@ def test_log_apply_evalf():
     assert value.epsilon_eq(Float("0.58496250072115618145373"))
 
 
+def test_log_leading_term():
+    p = Symbol('p')
+
+    # Test for STEP 3
+    assert log(1 + x + x**2).as_leading_term(x, cdir = 1) == x
+    # Test for STEP 4
+    assert log(2*x).as_leading_term(x, cdir = 1) == log(x) + log(2)
+    assert log(2*x).as_leading_term(x, cdir = -1) == log(x) + log(2)
+    assert log(-2*x).as_leading_term(x, cdir = 1, logx = p) == p + log(2) + I*pi
+    assert log(-2*x).as_leading_term(x, cdir = -1, logx = p) == p + log(2) - I*pi
+    # Test for STEP 5
+    assert log(-2*x + (3 - I)*x**2).as_leading_term(x, cdir = 1) == log(x) + log(2) - I*pi
+    assert log(2*x + (3 - I)*x**2).as_leading_term(x, cdir = -1) == log(x) + log(2) - 2*I*pi
```

Comment:
Is this supposed to be missing a '-' sign in front of 2*x?

Reply:
No ,the real essence of STEP 5 comes into picture for `log(2*x + (3 - I)*x**2)` through any negative `cdir`. Yeah it might look weird because I've added both tests(the upper one and this) which would go through STEP 5 and neglected the counterparts which would skip the step . I guess I'll add them in the next commit to make more sense , as those tests would then act as controls and would showcase cases which won't need STEP 5